LibDeflate.lua

```diff
@@ -1992,11 +1998,13 @@ local function Deflate(configs, WriteBits, WriteString, FlushWriter, str
 				string_table[j] = string_table[i-offset]
 				j = j + 1
             end
+            local delete = {}
 			for k, t in pairs(hash_tables) do
 				local tSize = #t
 				if tSize > 0 and block_end+1 - t[1] > 32768 then
-					if tSize == 1 then
-						--hash_tables[k] = nil -- seems to be causing problems
+                    if tSize == 1 then
+                        if _G.os and _G.os.pullEvent then table.insert(delete, k)
+						else hash_tables[k] = nil end
 					else
```

Comment:
Can you confirm it is a ComputerCraft specific workaround?

Reply:
`os.pullEvent` always exists on ComputerCraft, and does not seem to exist on anything else.

Reply:
Not going to merge changes which includes ComputerCraft specific API.
Caller should be able to workaround "os.pullEvent" by splitting big input into multiple parts and compress multiple times.